grpc/stream.go

```diff
@@ -284,6 +279,23 @@ func (s *stream) writeData(wg *sync.WaitGroup) {
 	}
 }
 
+func (s *stream) processSendError(err error) {
+	if errors.Is(err, io.EOF) {
+		s.logger.WithError(err).Debug("can't send data to the closed stream")
+
+		s.tq.Queue(func() error {
+			return s.closeWithError(nil)
+		})
+
+		return
+	}
+
+	s.tq.Queue(func() error {
+		return s.closeWithError(nil)
+	})
+	s.logger.WithError(err).Error("failed to send data to the stream")
```

Comment:
```suggestion
	s.logger.WithError(err).Error("failed to send data to the stream")
	s.tq.Queue(func() error {
		return s.closeWithError(err)
	})
```

I would always first log before doing this as otherwise it could just have more logs in between.

Also I expect you meant to add the `err` here?